ls1intum/Artemis
# `Team exercises`: Fix team participation table sometimes missing entries

### Checklist
#### General


#### Server
- No Changes
#### Client
- No significant changes

### Motivation and Context



`getAssessmentLink` is called with `participation` and/or `submission` having the value `null`, causing the errror:
```
TypeError: Cannot read properties of null (reading 'id')
    at rr.getAssessmentLink (team-participation-table.component.ts:138)
    ....
```
when a team didn't start an exercise. this then causes this and all following rows of the participation table to be skipped, making any submission among them **unassessable**.

The method should really never be called with `null` arguments in the first place, but i don't know enough about angular and how the artemis UI works to do that.

### To Reproduce
1. create a team execise in one course
2. view the team details / participations page, it should list the exercise, but it doesn't
3. if the team belong to more than one exercise, the list may show up partially
4. if an exercise that requires assessment comes after en exercise that was not participated in, it will not show up and be impossible to assess

### Description
Simply sets values for `participation` and `submission` so that no issues appear.
- set `participation` to an empty object, so that the id access succeeds and returns undefined, valid input for participationId in `getLinkToSubmissionAssessment`.
- set `submission` to `"new"`, so that the link creation can continue, the generated link will likely be nonsense, but the result will not be used anyways as the assessment button will be disabled when there s no submission.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 TA
- 1 Student

1. Log in to Artemis
2. Reproduce the bug as above, before updating the client
3. see that all exercises are shown now

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots



Before:
![artemis ase in tum de_courses_147_exercises_5276_teams_23151](https://user-images.githubusercontent.com/44437514/141292019-a254df60-2f9e-4f35-9cd7-ed55d14c8244.png)
After:
![localhost_8080_courses_147_exercises_5276_teams_23151](https://user-images.githubusercontent.com/44437514/141292033-f29294d0-24f9-4f65-bc37-98a26a34dca7.png)
